server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1713,7 +1711,7 @@ private static Checkpoint readCheckpoint(Path location, String expectedTranslogU
         } catch (TranslogCorruptedException ex) {
             throw ex; // just bubble up.
         } catch (Exception ex) {
-            throw new TranslogCorruptedException("Translog at [" + location + "] is corrupted", ex);
```

Comment:
@DaveCTurner We need to convert to a `TranslogCorruptedException` so that the recovery target can detect the error then switch from a sequence-based to a file-based recovery. See https://github.com/elastic/elasticsearch/pull/28587/files#diff-cdd186e2c33a5fa4d4200f9119684443L384